doc/api/fs.md

```diff
@@ -2716,10 +2716,14 @@ Returns an integer representing the file descriptor.
 For detailed information, see the documentation of the asynchronous version of
 this API: [`fs.open()`][].
 
-## `fs.read(fd, buffer, offset, length, position, callback)`
+## `fs.read(fd, [buffer[, [offset[, length[, position]]]], callback)`
```

Comment:
As much as I dislike polymorphic signatures, I'd much prefer the approach of moving to an `options` object as an alternative here. That is:

```js
fs.read(fd, { offset: n, position: n }, callback)
```

It accomplishes the same goal with a much cleaner API and implementation, without the ambiguity of which argument was meant to be passed in.

Reply:
I think i see what you are saying.  Just for my own clarification,  we keep the current signature of `fs.read(fd, buffer, offset, lenght, position, callback)`,  but then add another signature of `fs.read(fd, options, callback)` where the `options` is the buffer, offset, length, position params

So then if a user only wanted to specify some of the params, they would have to use the "options" object signature. 

Does that sound correct?

Reply:
@jasnell Something else that i thought of was that the `fs.write` method is similar, in that it allows for optional parameters to be passed without the use of an options object.  

Perhaps we don't go the options object route?  Or if we do, maybe that function and others(not sure if there are)should be updated also?

Reply:
I think moving that direction is ideal but doesn't have to be done all at once

Reply:
@jasnell @ronag does what i said here, https://github.com/nodejs/node/pull/31402#discussion_r371809733  makes sense?